### PR TITLE
CRAFTERCMS-1833

### PR DESCRIPTION
**Pull request by dejan-brkic**
Fixed issue with delete dependency cascade

### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/1833

### CODE REVIEW

File: src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java [hunk 1]
```diff
@@ -479,6 +482,82 @@ public void deleteSiteDependencies(@ValidateStringParam(name = "site") String si
...
+
+        // Step 2: collect all dependencies from DB
+        Set<String> dependencies = getItemDeleteDependencies(site, paths);
+        toRet.addAll(dependencies);
+        boolean doItAgain = false;
+
+        doItAgain = doItAgain || processedPaths.addAll(children);
+        doItAgain = doItAgain || processedPaths.addAll(dependencies);
+        if (doItAgain) {
+            List<String> pathsToProces = new ArrayList<String>();
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java]: pathsToProcess not PathstoProces

File: src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java [hunk 2]
```diff
@@ -479,6 +482,82 @@ public void deleteSiteDependencies(@ValidateStringParam(name = "site") String si
...
+            pathsToProces.addAll(children);
+            pathsToProces.addAll(dependencies);
+            toRet.addAll(getDeleteDependenciesInternal(site, pathsToProces, processedPaths));
+        }
+
+        // Step 4: recursion ???
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java]: Recursion is above in line 516, no? Why this comment?

File: src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java [hunk 3]
```diff
@@ -479,6 +482,82 @@ public void deleteSiteDependencies(@ValidateStringParam(name = "site") String si
...
+        return toRet;
+    }
+
+    private Set<String> getItemDeleteDependencies(String site, List<String> paths) {
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java]: Method name doesn't convey what method actually does. Should this be called getItemDependencies or getFilteredItemDependencies?

Also, this is the third level in the call stack. Why this complexity. It's becoming hard to follow.

File: src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java [hunk 4]
```diff
@@ -479,6 +482,82 @@ public void deleteSiteDependencies(@ValidateStringParam(name = "site") String si
...
+            filtered.addAll(deps);
+            filtered = filterItemSpecificDependencies(filtered);
+            exitCondition = !toRet.addAll(filtered);
+            paths.clear();
+            paths.addAll(deps);
+        } while (!exitCondition);
+        return toRet;
+    }
+
+    private Set<String> getAllChildrenRecursively(String site, List<String> paths) {
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java]: Doesn't this method exist somewhere else as some form of getTree? Or is this here because it returns a simple list of paths?

**Review by sumerjabri — CHANGES_REQUESTED** [review]
Ping me to discuss.

**Comment by dejan-brkic** [comment]
Added requested changes
